scripts/run.sh

```diff
@@ -26,243 +18,80 @@ source "$SUBNET_EVM_PATH"/scripts/versions.sh
-  BIN=${GOBIN}/avalanche-network-runner
-fi
-echo "launch avalanche-network-runner in the background"
-$BIN server \
-  --log-level debug \
-  --port=":12342" \
-  --grpc-gateway-port=":12343" &
-PID=${!}
+DATA_DIRS=()
+CMDS=()
+for (( i=0; i <5; i++ ))
+do
+  NODE_NAME=node$(($i+1))
+  NODE_DATA_DIR="$DATA_DIR/$NODE_NAME"
+  DATA_DIRS+=("$NODE_DATA_DIR")
+  echo "Creating data directory: $NODE_DATA_DIR"
+  mkdir -p $NODE_DATA_DIR
+  NODE_CONFIG_FILE_PATH="$NODE_DATA_DIR/config.json"
+  _set_config $NODE_CONFIG_FILE_PATH
+  
+  CMD="$AVALANCHEGO_PATH --data-dir=$NODE_DATA_DIR --config-file=$NODE_CONFIG_FILE_PATH"
+  if [ $i -gt 0 ]; then
+    echo "Adding CLI options for node$(($i+1))"
+    CMD="$CMD --staking-port=$((9651+2*$i)) --http-port=$((9650+2*$i)) --bootstrap-ips=127.0.0.1:9651 --bootstrap-ids=NodeID-7Xhw2mDxuDS44j42TCB6U5579esbSt3Lg"
+  else
+    CMD="$CMD --staking-tls-key-file-content=$STAKING_TLS_KEY_FILE_CONTENT --staking-tls-cert-file-content=$STAKING_TLS_CRT_FILE_CONTENT"
+  fi
+  
+  echo "Created command $CMD"
+  CMDS+=("$CMD")
+done
 
-run_ginkgo() {
-  echo "building e2e.test"
-  # to install the ginkgo binary (required for test build and run)
-  go install -v github.com/onsi/ginkgo/v2/ginkgo@${GINKGO_VERSION}
-  ginkgo -h
 
-  ACK_GINKGO_RC=true ginkgo build ./tests/e2e
+echo "Starting AvalancheGo network with the commands:"
+echo ""
+for (( i=0; i<5; i++ ))
+do
+  echo "CMD $i: ${CMDS[i]}"
+  echo ""
+done
 
-  # By default, it runs all e2e test cases!
-  # Use "--ginkgo.skip" to skip tests.
-  # Use "--ginkgo.focus" to select tests.
-  echo "running e2e tests with SKIP_NETWORK_RUNNER_START ${SKIP_NETWORK_RUNNER_START}"
-  ./tests/e2e/e2e.test \
-    --ginkgo.vv \
-    --network-runner-log-level debug \
-    --network-runner-grpc-endpoint="0.0.0.0:12342" \
-    --avalanchego-path=${AVALANCHEGO_PATH} \
-    --avalanchego-plugin-dir=${AVALANCHEGO_PLUGIN_DIR} \
-    --avalanchego-log-level=${AVALANCHE_LOG_LEVEL} \
-    --vm-genesis-path=$BASEDIR/genesis.json \
-    --output-path=$BASEDIR/avalanchego-${VERSION}/output.yaml \
-    --skip-network-runner-start=${SKIP_NETWORK_RUNNER_START} \
-    --skip-network-runner-shutdown=${SKIP_NETWORK_RUNNER_SHUTDOWN} \
-    --ginkgo.label-filter="${GINKGO_LABEL_FILTER}"
+# cleanup sends SIGINT to each tracked process
+function cleanup_process_group(){
+  echo "Terminating AvalancheGo network process group"
+  kill 0 # This kills a process group rather than a process with the ID 0, used to kill subshell to cleanup below
 }
 
-run_simulator() {
-  #################################
-  echo "building simulator"
-  pushd ./cmd/simulator
-  go install -v .
-  popd
-
-  echo "running simulator"
-  simulator \
-  --cluster-info-yaml=$BASEDIR/avalanchego-${VERSION}/output.yaml \
-  --keys=./cmd/simulator/.simulator/keys \
-  --timeout=30s \
-  --concurrency=10 \
-  --base-fee=25 \
-  --priority-fee=1
+function execute_cmd() {
+  echo "Executing command: $@"
+  $@
 }
 
+execute_cmd ${CMDS[0]}
```

Comment:
Could we finish the TODO below as part of this PR? (not sure why we only care about CMDS[0] here)

Reply:
There seems to be a bug with staking disabled that causes multi-node networks to not start consensus on subnets. We already fixed a bug in AvalancheGo where subnets would not start with a single node non-staking network, but I have not figured out what the issue is with multiple nodes yet.

Reply:
Removing this in favor of using other tools to set this up.